contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,508 @@
+                )
+            ]
+        );
+
+        let res = query(deps.as_ref(), env.clone(), QueryMsg::LatestStage {}).unwrap();
+        let latest_stage: LatestStageResponse = from_binary(&res).unwrap();
+        assert_eq!(1u8, latest_stage.latest_stage);
+
+        let res = query(
+            deps.as_ref(),
+            env,
+            QueryMsg::MerkleRoot {
+                stage: latest_stage.latest_stage,
+            },
+        )
+        .unwrap();
+        let merkle_root: MerkleRootResponse = from_binary(&res).unwrap();
+        assert_eq!(
+            "634de21cde1044f41d90373733b0f0fb1c1c71f9652b905cdf159e73c4cf0d37".to_string(),
+            merkle_root.merkle_root
+        );
+    }
+
+    #[test]
+    fn claim() {
+        let mut deps = mock_dependencies(&[]);
+
+        let msg = InstantiateMsg {
+            owner: "owner0000".to_string(),
+            cw20_token_address: "token0000".to_string(),
+        };
+
+        let env = mock_env();
+        let info = mock_info("addr0000", &[]);
+        let _res = instantiate(deps.as_mut(), env, info, msg).unwrap();
+
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::RegisterMerkleRoot {
+            merkle_root: "5d4f48f147cb6cb742b376dce5626b2a036f69faec10cd73631c791780e150fc"
+                .to_string(),
+        };
+        let _res = execute(deps.as_mut(), env, info, msg).unwrap();
+
+        let msg = ExecuteMsg::Claim {
+            amount: Uint128::from(10220u128),
+            stage: 1u8,
+            proof: vec![
+                "561b09601236f43b0065d1ea601df754989911ed6da16ba8afed62c458f86352".to_string(),
+                "280777995d054081cbf208bccb70f8d736c1766b81d90a1fd21cd97d2d83a5cc".to_string(),
+                "3946ea1758a5a2bf55bae1186168ad35aa0329805bc8bff1ca3d51345faec04a".to_string(),
+            ],
+        };
+
+        let env = mock_env();
+        let info = mock_info("wasm1qzy8rg0f406uvvl54dlww6ptlh30303xq2u3xu", &[]);
+        let res = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone()).unwrap();
+        let expected = SubMsg::new(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: "token0000".to_string(),
+            funds: vec![],
+            msg: to_binary(&Cw20ExecuteMsg::Transfer {
+                recipient: "wasm1qzy8rg0f406uvvl54dlww6ptlh30303xq2u3xu".into(),
+                amount: Uint128::from(10220u128),
+            })
+            .unwrap(),
+        }));
+        assert_eq!(res.messages, vec![expected]);
+
+        assert_eq!(
+            res.attributes,
+            vec![
+                attr("action", "claim"),
+                attr("stage", "1"),
+                attr("address", "wasm1qzy8rg0f406uvvl54dlww6ptlh30303xq2u3xu"),
+                attr("amount", "10220")
+            ]
+        );
+
+        /*
+        TODO enable after 0.16 release https://github.com/CosmWasm/cosmwasm/pull/989/files
```

Comment:
`TODO` => `TODO:` - will be recognised by most IDEs ;)

Reply:
Good tip